data/importToDb.js

```diff
@@ -11,28 +11,38 @@ DatabaseManager.establishConnection().then(() => {
         DatabaseManager.getDatabase().beginTransaction();
 
         // FIXME: The driver does not support prepared statements, yet. Rewrite once it is supported.
-        const rawData = require('./rawData-frankfurt.geo');
+        const rawData = require('./rawData-bw.geo');
         DatabaseManager.getDatabase().query('-- TRUNCATE TABLE trash_bins;', err => {
             console.log('Truncated table');
             if (err) throw err;
             const sql = 'INSERT IGNORE INTO trash_bins(trashBinID, latitude, longitude, data) VALUES (null, ?, ?, ?);';
 
-            successfulRequests = 0;
+            let successfulRequests = 0;
 
-            console.time('import');
+            // improve performance tremendously by saving how big our array is
+            const expectedLength = rawData.elements.length;
+            const percentageStep = 100 / expectedLength;
+            console.log(`Importing ${expectedLength} trash bins.`);
+
+            // start measuring performance before entering the loop
+            console.time('Import');
             for (const entry of rawData.elements) {
                 DatabaseManager.queryPromisify(sql, [
                     entry.lat,
                     entry.lon,
                     JSON.stringify(entry.tags)
                 ]).then(() => {
-                    if (++successfulRequests === rawData.elements.length) {
-                        process.stdout.write(os.EOL);
-                        console.timeEnd('import');
-                        DatabaseManager.getDatabase().commit(() => process.exit());
-                    }
-                    if (successfulRequests % 5 === 0) {
-                        process.stdout.write('.');
+                    if (++successfulRequests % 100 === 0 || successfulRequests === expectedLength) {
+                        // print out current percentage with 2 decimals
+                        const currentPercentage = Math.round(percentageStep * successfulRequests * 100) / 100;
+                        process.stdout.write(`${successfulRequests}… (${currentPercentage}%)${os.EOL}`);
+
+                        if (successfulRequests === expectedLength) {
```

Comment:
this alone makes it approx. 500ms faster with ~12,500 entries because we only do this check each 100 times, so not causing a noticeable delay.